pkg/config/setup/config.go

```diff
@@ -1222,6 +1222,9 @@ func telemetry(config pkgconfigmodel.Setup) {
 	// Agent Telemetry. It is experimental feature and is subject to change.
 	// It should not be enabled unless prompted by Datadog Support
 	config.BindEnvAndSetDefault("agent_telemetry.enabled", false)
+	config.SetKnown("agent_telemetry.additional_endpoints.*")
+	bindEnvAndSetLogsConfigKeys(config, "agent_telemetry.")
```

Comment:
I have the impression that this function `bindEnvAndSetLogsConfigKeys` is seetting way more keys that we currently use in agenttelemetry. 

```
config.BindEnv(prefix + "logs_dd_url") // Send the logs to a proxy. Must respect format '<HOST>:<PORT>' and '<PORT>' to be an integer
	config.BindEnv(prefix + "dd_url")
	config.BindEnv(prefix + "additional_endpoints")
	config.BindEnvAndSetDefault(prefix+"use_compression", true)
	config.BindEnvAndSetDefault(prefix+"compression_level", 6) // Default level for the gzip/deflate algorithm
	config.BindEnvAndSetDefault(prefix+"batch_wait", DefaultBatchWait)
	config.BindEnvAndSetDefault(prefix+"connection_reset_interval", 0) // in seconds, 0 means disabled
	config.BindEnvAndSetDefault(prefix+"logs_no_ssl", false)
	config.BindEnvAndSetDefault(prefix+"batch_max_concurrent_send", DefaultBatchMaxConcurrentSend)
	config.BindEnvAndSetDefault(prefix+"batch_max_content_size", DefaultBatchMaxContentSize)
	config.BindEnvAndSetDefault(prefix+"batch_max_size", DefaultBatchMaxSize)
	config.BindEnvAndSetDefault(prefix+"input_chan_size", DefaultInputChanSize) // Only used by EP Forwarder for now, not used by logs
	config.BindEnvAndSetDefault(prefix+"sender_backoff_factor", DefaultLogsSenderBackoffFactor)
	config.BindEnvAndSetDefault(prefix+"sender_backoff_base", DefaultLogsSenderBackoffBase)
	config.BindEnvAndSetDefault(prefix+"sender_backoff_max", DefaultLogsSenderBackoffMax)
	config.BindEnvAndSetDefault(prefix+"sender_recovery_interval", DefaultForwarderRecoveryInterval)
	config.BindEnvAndSetDefault(prefix+"sender_recovery_reset", false)
	config.BindEnvAndSetDefault(prefix+"use_v2_api", true)
	config.SetKnown(prefix + "dev_mode_no_ssl")
```

Should we simply manually add the ones that we are currently supporting?

Reply:
I found that Viper generates warnings that these keys in configuration are missing when I have used that endpoints collection API. This is what other clients which using the API have added.

Reply:
Right.

It seems that we are adding a lot of configuration keys so we can reuse some code from the logs agent component. Not saying that we do, but what we instead duplicate the logic we care inside the agenttelemetry component and avoid adding this many unused configuration keys? 

Also, it seems that you have a comment the mention that you planning on removing the usage:
> 	// borrowed and styled after EP Forwarder newHTTPPassthroughPipeline().
	// Will be eliminated in the future after switching to EP Forwarder.

Reply:
That what EP Forwarder clients, at least most of them appears to do, https://github.com/search?q=org%3ADataDog%20bindEnvAndSetLogsConfigKeys&type=code (meaning if I move to EP Forwarder I may repeat it too).
```
387 | bindEnvAndSetLogsConfigKeys(config, "network_devices.metadata.")
430 | bindEnvAndSetLogsConfigKeys(config, "network_devices.snmp_traps.forwarder.")
447 | bindEnvAndSetLogsConfigKeys(config, "network_devices.netflow.forwarder.")
459 | bindEnvAndSetLogsConfigKeys(config, "network_path.forwarder.")
805 | bindEnvAndSetLogsConfigKeys(config, "container_lifecycle.")
809 | bindEnvAndSetLogsConfigKeys(config, "container_image.")
816 | bindEnvAndSetLogsConfigKeys(config, "sbom.")
840 | bindEnvAndSetLogsConfigKeys(config, "service_discovery.forwarder.")
```
I think that it is a small evil compare from code duplication (and error introduction). Besides, why not to be able control how agent telemetry payloads are transfered (transport-wise), meaning I do not see any contradiction in principle.
 
What @gh123man and @remeh who know a thing or two about EP Forwarder would suggest?

Reply:
@GustavoCaso I think we have established that additional_endpoints are beneficial for short and long term agent_telemetry future. If you have no other concern can you please approve the change?

Reply:
If the plan is to move this into epforwarder, I think leaving it as-is is fine since you will inherit many of the other config options as a result. 

> Also, the fact that the agenttelemtry component imports "github.com/DataDog/datadog-agent/comp/logs/agent/config" feels a bit off.

Sadly, this is an artifact of the past. The "logs" pipeline is really the events platform pipeline. But it's original use case was for logs. Someday this whole package should be renamed.

Reply:
@iglendd 

> I think we have established that additional_endpoints are beneficial for short and long term agent_telemetry future

Absolutely. 

Just to clarify my comment. I wasn't saying not to use additional endpoints, I was concern that we were adding my configuration option for the `agent_telemetry` namespace, due to the fact that we use `bindEnvAndSetLogsConfigKeys`.